template/before.sh.erb

```diff
@@ -223,7 +223,7 @@ sed 's/^ \{2\}//' > "${PYTHON_WRAPPER_FILE}" << EOL
 
   # Setup PySpark
   export PYTHONPATH="\${SPARK_HOME}/python:\$PYTHONPATH"
-  export PYTHONPATH="\${SPARK_HOME}/python/lib/<%= py4j %>:\$PYTHONPATH"
+  export PYTHONPATH="<%= py4j %>:\$PYTHONPATH"
```

Comment:
If you have simlinks then you can just use them directly here and we don't need that case statement. I think that's the whole value. We won't have to worry about the `py4j` computation and we can just add the symlink to the PYTHONPATH.

```
 export PYTHONPATH="\${SPARK_HOME}/python/lib/py4j-src.zip:\$PYTHONPATH"
```
I don't know where SPARK_PY4J_PATH is being set, but if it's in the module and ZQ says it'll be stable to use, then we can use _it_ instead.

```
export PYTHONPATH="\$SPARK_PY4J_PATH:\$PYTHONPATH"
```

Reply:
> If you have simlinks then you can just use them directly here and we don't need that case statement. I think that's the whole value. We won't have to worry about the `py4j` computation and we can just add the symlink to the PYTHONPATH.

I just wanted to test that the selection of version 3.0.1 worked for you as well before making the change to rely only on the context and remove the hard-coding. Happy to have ZQ finish the remaining symlinks and remove the case statements.

> I don't know where SPARK_PY4J_PATH is being set, but if it's in the module and ZQ says it'll be stable to use, then we can use _it_ instead.

That env var is what ZQ setup for the symlinks, so you end up with the following in a shell for example:
<pre>
module load spark/3.0.1
ls $SPARK_PY4J_PATH
/usr/local/spark/3.0.1/python/lib/py4j-0.10.9-src.zip
</pre>

So I'll let him know to finish, make the changes when he's done, and test the context works correctly for the versions.

Reply:
Oh i see. Yea ok I can test it out.